lib/index.ts

```diff
@@ -1,167 +1,90 @@
 import 'source-map-support/register';
 import * as fs from 'fs';
 import * as path from 'path';
-import * as _ from 'lodash';
-
-enum DepType {
-  prod = 'prod',
-  dev = 'dev',
-}
-
-interface PkgTree {
-  name: string;
-  version: string;
-  dependencies?: {
-    [dep: string]: PkgTree;
-  };
-  depType?: DepType;
-  hasDevDependencies?: boolean;
-  cyclic?: boolean;
-}
-
-interface ManifestFile {
-  name?: string;
-  dependencies?: {
-    [dep: string]: string;
-  };
-  devDependencies?: {
-    [dep: string]: string;
-  };
-  version?: string;
-}
-
-interface Lockfile {
-  name: string;
-  version: string;
-  dependencies?: LockfileDeps;
-}
-
-interface LockfileDeps {
-  [depName: string]: LockfileDep;
-}
-
-interface LockfileDep {
-  version: string;
-  requires?: {
-    [depName: string]: string;
-  };
-  dependencies?: LockfileDeps;
-  dev?: boolean;
+import {LockfileParser, Lockfile, ManifestFile, PkgTree, DepType} from './parsers/parser';
+import {PackageLockParser} from './parsers/package-lock-parser';
+import {YarnLockParser} from './parsers/yarn-lock-parse';
+import {getRuntimeVersion} from './utils';
+
+enum LockfileType {
+  npm = 'npm',
+  yarn = 'yarn',
 }
 
 export {
   buildDepTree,
   buildDepTreeFromFiles,
   PkgTree,
   DepType,
+  LockfileType,
 };
 
 async function buildDepTree(
-  manifestFileContents: string, lockFileContents: string, includeDev = false): Promise<PkgTree> {
-
-  const lockFile: Lockfile = JSON.parse(lockFileContents);
-  const manifestFile: ManifestFile = JSON.parse(manifestFileContents);
+  manifestFileContents: string, lockFileContents: string,
+  includeDev = false, lockfileType?: LockfileType): Promise<PkgTree> {
 
-  if (!manifestFile.dependencies && !includeDev) {
-    throw new Error("No 'dependencies' property in package.json");
+  if (!lockfileType) {
```

Comment:
I would prefer we don't assume a type but throw, unexpected errors during parsing a wring file type will be worse than use being extra strict and provide back a very clear useful message.